src/Microsoft.OData.Core/JsonLight/ODataJsonLightReaderNestedResourceInfo.cs

```diff
@@ -82,7 +79,16 @@ private ODataJsonLightReaderNestedResourceInfo(ODataNestedResourceInfo nestedRes
 
             this.nestedProperty = nestedProperty;
             this.nestedResourceInfo = nestedResourceInfo;
-            this.nestedResourceType = nestedResourceType;
+
+            bool resourceTypeNullable = true;
+            if (nestedProperty != null && nestedProperty.Type != null)
+            {
+                resourceTypeNullable = nestedProperty.Type.IsNullable;
+            }
+
+            IEdmStructuredType nestedType = (nestedResourceType != null) ? nestedResourceType : (nestedProperty != null) ? nestedProperty.Type.ToStructuredType() : null;
+
+            NestedResourceTypeReference = (nestedType != null) ? nestedType.ToTypeReference(resourceTypeNullable).AsStructured() :  null;

```

Comment:
Just do like:


if (nestedProperty != null)
{
                NestedResourceTypeReference = nestedProperty.Type.AsStructured();
}
else
 {
       NestedResourceTypeReference = nestedResourceType.ToTypeReference(true);
 }

Reply:
This is what I tried first but a bunch of test cases failed. 
I can clean up the coding style but the logic will slightly be different. 

---
In reply to: [267147384](https://github.com/OData/odata.net/pull/1322#discussion_r267147384) [](ancestors = 267147384)